python/sglang/srt/entrypoints/verl_engine.py

```diff
@@ -39,11 +41,26 @@ def __init__(
         node_rank = self._tp_rank // tp_size_per_node
         first_rank_in_node = self._tp_rank % tp_size_per_node == 0
 
-        if first_rank_in_node:
+        if first_rank_in_node and "launch_server" not in kwargs:
             os.environ["SGLANG_BLOCK_NONZERO_RANK_CHILDREN"] = "0"
             self._engine = Engine(
                 **kwargs, tp_size=self._tp_size, node_rank=node_rank, nnodes=nnodes
             )
+        elif "launch_server" in kwargs and kwargs["launch_server"]:
+            del kwargs["launch_server"]
+            if "server_args" in kwargs:
+                # Directly load server_args
+                server_args = kwargs["server_args"]
+            else:
+                # Construct server_args from kwargs
+                if "log_level" not in kwargs:
+                    # Do not print logs by default
+                    kwargs["log_level"] = "error"
+                server_args = ServerArgs(**kwargs)
```

Comment:
Hmm looks like the code is copied from Engine, if so then

* it would be great to avoid copying code, e.g. extract a function `compute_server_args_from_kwargs`
* the layering is a little bit weird if we keep it: the same logic, "computing serverargs from kwargs", is put (a) in Engine/Adapter layer, and (b) in the VerlEngine layer which is one layer above Engine/Adapter. thus would be great to move it to Adapter.